parser.go

```diff
@@ -98,6 +98,13 @@ func (p *Parser) parseExpression(stop ...rune) (Expression, error) {
 			continue
 		}
 
+		// Ignore empty variable references
+		if strings.HasPrefix(p.input[p.pos:], `$ `) {
+			p.pos += 2
+			expr = append(expr, ExpressionItem{Text: `$ `})
+			continue
+		}
```

Comment:
This is a bandaid fix for a specific case - arguably we should be treating a `$` as a literal if there is no next character, or the next character is anything we don't parse as an expansion later on. Something like:

```go
// If we run into a dollar sign, what we do with it depends on what comes next
if c == '$' {
	if p.pos == len(p.input)-1 {
		// It can't be an expansion.
		p.pos++
		expr = append(expr, ExpressionItem{Text: "$"})
		continue
	}

	if c2 := p.peekRune(); c2 == '{' || unicode.IsLetter(c2) {
		// It's ${ or $something ... probably an expansion!
		expansion, err := p.parseExpansion()
		if err != nil {
			return nil, err
		}
		expr = append(expr, ExpressionItem{Expansion: expansion})
		continue
	}
}
```

Note, though, that this only doesn't work because `peekRune` can't peek the rune after the current position...